### PR TITLE
update minitest and ruby version

### PR DESCRIPTION
**Pull request by HParker**
All the tests still pass with 2.3.1, (yay). How do you feel about making 2.3.1 the `.ruby-version` for this project?

### CODE REVIEW

**Comment by simeonwillbanks** [comment]
> How do you feel about making 2.3.1 the .ruby-version for this project?

Yes, thank you!  🍻

File: .travis.yml [hunk 1]
```diff
@@ -2,3 +2,4 @@ language: ruby
 rvm:
   - 2.1.0
   - 2.1.1
+  - 2.3.1
```
Comment (simeonwillbanks) [inline .travis.yml]: While we are here, why don't we please clean this up a bit.

```yml
language: ruby
rvm:
- 2.1
- 2.2
- 2.3
matrix:
  allow_failures:
  - rvm: ruby-head
```

I [validated](http://lint.travis-ci.org/) this YAML with Travis WebLint.

**Comment by HParker** [comment]
Huh, I am a little confused why the build is angry.

Is the gemspec now invalid somehow?

**Comment by simeonwillbanks** [comment]
@HParker That is strange.  I tried [rebuilding](https://travis-ci.org/simeonwillbanks/busted/jobs/161821269), and it still failed.  I'll try [`ruby 2.1.5p273`](https://travis-ci.org/simeonwillbanks/busted/jobs/161821269#L132) in my local environment.

**Comment by simeonwillbanks** [comment]
@HParker I was able to replicate in my local environment. 

```
~/Code/busted update_minitest*
❯ ruby --version
ruby 2.1.5p273 (2014-11-13 revision 48405) [x86_64-darwin14.0]

~/Code/busted update_minitest*
❯ gem --version
2.6.6

~/Code/busted update_minitest*
❯ bundle --version
Bundler version 1.7.6

~/Code/busted update_minitest*
❯ bundle install --jobs=3 --retry=3
Fetching gem metadata from https://rubygems.org/..............
Resolving dependencies...
Using bundler 1.7.6

NoMethodError: undefined method `spec' for nil:NilClass
Installing rake 11.3.0
An error occurred while installing busted (0.2.3), and Bundler cannot continue.
Make sure that `gem install busted -v '0.2.3'` succeeds before bundling.
```

It looks like this was an [issue with bundler](https://github.com/bundler/bundler/issues/3558).

I forced the latest Ruby 2.1.x, and that brought in a new bundler.  Now it's bundling!

```
~/Code/busted update_minitest*
❯ ruby --version
ruby 2.1.10p492 (2016-04-01 revision 54464) [x86_64-darwin15.0]

~/Code/busted update_minitest*
❯ gem --version
2.6.6

~/Code/busted update_minitest*
❯ bundle --version
Bundler version 1.13.1

~/Code/busted update_minitest*
❯ bundle install --jobs=3 --retry=3
Fetching gem metadata from https://rubygems.org/.............
Fetching version metadata from https://rubygems.org/.
Resolving dependencies...
Using bundler 1.13.1
Using busted 0.2.3 from source at `.`
Installing minitest 5.9.0
Installing rake 11.3.0
Bundle complete! 4 Gemfile dependencies, 4 gems now installed.
Use `bundle show [gemname]` to see where a bundled gem is installed.
```

The tests are passing...

```
~/Code/busted update_minitest*  30s
❯ rake
/Users/simeonwillbanks/Code/busted/lib/busted/counter.rb:33: warning: private attribute?
/Users/simeonwillbanks/Code/busted/lib/busted/counter.rb:33: warning: private attribute?
/Users/simeonwillbanks/Code/busted/lib/busted/tracer.rb:40: warning: private attribute?
/Users/simeonwillbanks/Code/busted/lib/busted/tracer.rb:41: warning: private attribute?
/Users/simeonwillbanks/Code/busted/test/busted_test.rb:66: warning: possibly useless use of + in void context
/Users/simeonwillbanks/Code/busted/test/busted_test.rb:286: warning: ambiguous first argument; put parentheses or even spaces
/Users/simeonwillbanks/Code/busted/test/busted_test.rb:297: warning: ambiguous first argument; put parentheses or even spaces
Run options: --seed 58520

# Running:

...............
.
......................................

Fabulous run in 4.738791s, 11.3953 runs/s, 17.0930 assertions/s.

54 runs, 81 assertions, 0 failures, 0 errors, 0 skips
```

I'll update the `.travis.yml` and push.

**Comment by simeonwillbanks** [comment]
@HParker We are 💚 💯 🎉 .  Look good to you?

**Comment by HParker** [comment]
Woo! Thanks for your help!

Weird that specifying 2.1 doesn't pick the latest 2.1.

**Comment by simeonwillbanks** [comment]
@HParker you're welcome, and thanks for moving busted forward.  ❤️

**Pull request merged**